manager/src/main/java/io/gingersnapproject/Caches.java

```diff
@@ -5,40 +5,98 @@
+               // TODO: need to revisit this later
+               int size = k.length() * 2 + 1;
+               if (v instanceof UniItem<String> uniItem) {
+                  var actualValue = uniItem.getItem();
+                  size += actualValue == null ? 0 : actualValue.length() * 2 + 1;
+                  if (size < 0) {
+                     size = Integer.MAX_VALUE;
+                  }
+               }
+               return size;
+            })
+            .build(key -> {
+               Uni<String> dbUni = databaseHandler.select(name, key)
+                     // Make sure to use memoize, so that if multiple people subscribe to this it won't cause
+                     // multiple DB lookups
+                     .memoize().indefinitely();
+               // This will replace the pending Uni from the DB with a UniItem so we can properly size the entry
+               // Note due to how lazy subscribe works the entry won't be present in the map  yet
+               dbUni.subscribe()
+                     .with(result -> replace(name, key, dbUni, result), t -> actualRemove(name, key, dbUni));
+               return dbUni;
+            }));
    }
 
-   public byte[] get(String name, String key) {
-      WrappedByteArray value = getOrCreateMap(name).getIfPresent(key);
-      return value == null ? null : value.getBytes();
+   private void replace(String name, String key, Uni<String> prev, String value) {
+      var cache = maps.get(name);
+      if (cache != null) {
+         cache.asMap().replace(key, prev, UniItem.fromItem(value));
+      }
+   }
+
+   private void actualRemove(String name, String key, Uni<String> value) {
+      var cache = maps.get(name);
+      if (cache != null) {
+         cache.asMap().remove(key, value);
+      }
+   }
+
+   public void put(String name, String key, String value) {
+      getOrCreateMap(name)
+            .put(key, UniItem.fromItem(value));
+   }
+
+   public Uni<String> get(String name, String key) {
+      return getOrCreateMap(name).get(key);
    }
 
    public Stream<String> getKeys(String name) {
-      Cache<String, WrappedByteArray> cache = getOrCreateMap(name);
+      Cache<String, ?> cache = getOrCreateMap(name);
       return cache.asMap()
             .keySet()
             .stream();
    }
 
-   public ConcurrentMap<String, Cache<String, WrappedByteArray>> getMaps() {
-      return maps;
+   public boolean remove(String name, String key) {
+      // TODO: technically this says it removed something even if the Uni contained a null value prior
+      // We put a null uni into the map if a value existed before. This way we cache the tombstone.
+      return getOrCreateMap(name).asMap()
+            .replace(key, Uni.createFrom().nullItem()) != null;
    }
 
-   public ConcurrentMap<String, Cache<String, List<byte[]>>> getMultiMaps() {
-      return multiMaps;
+   // Returns an estimated size which includes tombstone values
+   public long estimatedSize(String name) {
```

Comment:
There's no harm having this, but did you have a specific use-case in mind where we need to have this available via HotRod?